src/Microsoft.Health.Fhir.CosmosDb/Features/Search/FhirCosmosSearchService.cs

```diff
@@ -57,6 +57,12 @@ public class FhirCosmosSearchService : SearchService
             SearchOptions searchOptions,
             CancellationToken cancellationToken)
         {
+            // we're going to mutate searchOptions, so clone it first so the caller of this method does not see the changes.
+            searchOptions = searchOptions.Clone();
```

Comment:
This looks like a shallow clone.  If so any reference properties would be shared between the original object and the clone.  Do we alter the expression object that is passed to the Visitor, or de we generate a new object which is assigned to the clone?

Reply:
The expressions are all immutable so this should be safe